projects/RabbitMQ.Client/client/impl/ChannelBase.cs

```diff
@@ -193,24 +184,80 @@ protected void TakeOver(ChannelBase other)
 
         public void Close(ushort replyCode, string replyText, bool abort)
         {
-            _ = CloseAsync(new ShutdownEventArgs(ShutdownInitiator.Application, replyCode, replyText), abort);
-        }
-
-        private async Task CloseAsync(ShutdownEventArgs reason, bool abort)
-        {
+            var reason = new ShutdownEventArgs(ShutdownInitiator.Application, replyCode, replyText);
             var k = new ShutdownContinuation();
             ChannelShutdown += k.OnConnectionShutdown;
 
             try
             {
                 ConsumerDispatcher.Quiesce();
+
                 if (SetCloseReason(reason))
                 {
-                    _Private_ChannelClose(reason.ReplyCode, reason.ReplyText, 0, 0);
+                    _Private_ChannelClose(reason.ReplyCode, reason.ReplyText, reason.ClassId, reason.MethodId);
                 }
 
                 k.Wait(TimeSpan.FromMilliseconds(10000));
-                await ConsumerDispatcher.WaitForShutdownAsync().ConfigureAwait(false);
+
+                ConsumerDispatcher.WaitForShutdown();
+            }
+            catch (AlreadyClosedException)
+            {
+                if (!abort)
```

Comment:
This catch-!abort pattern is used quite often in this PR, there might be a few improvements here. In this specific case, you also catch(Exception)  and the bodies are all the same, so you can simply remvoe the other catches. I understand if you add them for visibility though (as the ones you specifically catch are "kinda" expected).

In general you can also do an exception filter, like this:
```catch(AlreadyClosedException) when (abort)```
This would reduce source code length of these blocks and improve readability.
Exception filters are rarely used in c#, but are kinda designed for this job, namely deciding if you want to enter the catch block prior to executing the catch block.

Reply:
Noted! https://github.com/rabbitmq/rabbitmq-dotnet-client/issues/1413